sympy/solvers/ode/tests/test_ode.py

```diff
@@ -2883,6 +2883,43 @@ def test_separable_reduced():
     assert sol == Eq(log(x**2*f(x))/2 - log(x**2*f(x) - 2)/2, C1 + log(x))
     assert checkodesol(eq, sol, order=1, solve_for_func=False)[0]
 
+    eq = Eq(f(x).diff(x) + f(x)/x * (1 + (x**(S(2)/3)*f(x))**2), 0)
+    # FIXME: This one hangs
+    #sol = dsolve(eq, hint = 'seprable_reduced')
+    # assert checkodesol(eq, sol) = (True, 0)
+
+    eq = Eq(f(x).diff(x) + f(x)/x * (1 + (x*f(x))**2), 0)
+    sol = dsolve(eq, hint = 'separable_reduced')
+    assert sol == [Eq(f(x), -sqrt(2)*sqrt(1/(C1 + log(x)))/(2*x)),\
+                   Eq(f(x), sqrt(2)*sqrt(1/(C1 + log(x)))/(2*x))]
+    assert checkodesol(eq, sol) == [(True, 0)]*2
+
+    eq = Eq(f(x).diff(x) + (x**4*f(x)**2 + x**2*f(x))*f(x)/(x*(x**6*f(x)**3 + x**4*f(x)**2)), 0)
+    sol = dsolve(eq, hint = 'separable_reduced')
+    assert sol == Eq(f(x), C1 + 1/(2*x**2))
+    assert checkodesol(eq, sol) == (True, 0)
+
+    eq = Eq(f(x).diff(x) + (f(x)**2)*f(x)/(x), 0)
+    sol = dsolve(eq, hint = 'separable_reduced')
+    assert sol == [Eq(f(x), -sqrt(2)*sqrt(1/(C1 + log(x)))/2),\
+                  Eq(f(x), sqrt(2)*sqrt(1/(C1 + log(x)))/2)]
+
+    eq = Eq(f(x).diff(x) + (f(x)+3)*f(x)/(x*(f(x)+2)), 0)
+    # FIXME: This one hangs
+    #sol == dsolve(eq, hint = 'separable_reduced')
+    #assert checkodesol(eq, sol) == (True, 0)
+
+    eq = Eq(f(x).diff(x) + (f(x)+3)*f(x)/x, 0)
+    sol = dsolve(eq, hint = 'separable_reduced')
+    assert sol == Eq(f(x), 3/(C1*x**3 - 1))
+    assert checkodesol(eq, sol) == (True, 0)
+
+    eq = Eq(f(x).diff(x) + (f(x)**2+f(x))*f(x)/(x), 0)
+    sol = dsolve(eq, hint='separable_reduced')
+    assert sol = Eq(-log(x) - log(f(x) + 1) + log(f(x)) + 1/f(x), C1)
+    # FIXME: Raise NotImplementedError
```

Comment:
```julia
In [57]: eq = Eq(f(x).diff(x) + (f(x)**2+f(x))*f(x)/(x), 0)                                          

In [58]: dsolve(eq, hint='separable_reduced')                                                        
Out[58]: 
                                       1       
-log(x) - log(f(x) + 1) + log(f(x)) + ──── = C₁
                                      f(x)     

In [59]: checkodesol(eq, dsolve(eq))                                                                 
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-59-1e7b692db659> in <module>
----> 1 checkodesol(eq, dsolve(eq))

~/Documents/git/sympy/sympy/solvers/ode/subscheck.py in checkodesol(ode, sol, func, order, solve_for_func)
    131     solved = sol.lhs == func and not sol.rhs.has(func)
    132     if solve_for_func and not solved:
--> 133         rhs = solve(sol, func)
    134         if rhs:
    135             eqs = [Eq(func, t) for t in rhs]

~/Documents/git/sympy/sympy/solvers/solvers.py in solve(f, *symbols, **flags)
   1187     ###########################################################################
   1188     if bare_f:
-> 1189         solution = _solve(f[0], *symbols, **flags)
   1190     else:
   1191         solution = _solve_system(f, symbols, **flags)

~/Documents/git/sympy/sympy/solvers/solvers.py in _solve(f, *symbols, **flags)
   1768 
   1769     if result is False:
-> 1770         raise NotImplementedError('\n'.join([msg, not_impl_msg % f]))
   1771 
   1772     if flags.get('simplify', True):

NotImplementedError: multiple generators [_X0, log(_X0 + 1), log(_X0)]
No algorithms are implemented to solve equation -C1 - log(_X0) + log(x) + log(_X0 + 1) - 1/_X0

```

Reply:
The error comes from `solve` inside `checkodesol` so we can use
```julia
In [40]: checkodesol(eq, sol, solve_for_func=False)                                                                               
Out[40]: (True, 0)
```
Maybe `checkodesol` should catch `NotImplementedError` from `solve` and continue checking in other ways even is `solve_for_func` us True (especially since that's the default).